Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -438,6 +454,33 @@ extension TLSConfiguration {
                                 applicationProtocols: [],
                                 shutdownTimeout: .seconds(5),
                                 keyLogCallback: nil,
+                                sniCallback: nil,
+                                renegotiationSupport: .none,
+                                additionalTrustRoots: [])
+    }
+    
+    /// Create a TLS configuration for dynamically generating server-size contexts with SNI.
+    ///
+    /// This provides sensible defaults while requiring that you provide any data that is necessary
+    /// for server-side function. For client use, try `makeClientConfiguration` instead.
+    ///
+    /// For customising fields, modify the returned TLSConfiguration object.
+    public static func makeServerConfiguration(
```

Comment:
I don't think we need this function: users can simply call the main version and set the SNI callback as desired. The advantage of the main callback is that it encourages users to set "fallback" certs and keys.